protocol/rest/server/server_impl/go_restful_server.go

```diff
@@ -82,23 +83,39 @@ func (grs *GoRestfulServer) Start(url common.URL) {
 
 // Publish a http api in go-restful server
 // The routeFunc should be invoked when the server receive a request
-func (grs *GoRestfulServer) Deploy(restMethodConfig *config.RestMethodConfig, routeFunc func(request server.RestServerRequest, response server.RestServerResponse)) {
-	ws := &restful.WebService{}
+func (grs *GoRestfulServer) Deploy(restServiceConfig *config.RestServiceConfig, invoker protocol.Invoker) {
+	/*ws := &restful.WebService{}
 	rf := func(req *restful.Request, resp *restful.Response) {
 		routeFunc(NewGoRestfulRequestAdapter(req), resp)
 	}
 	ws.Path(restMethodConfig.Path).
 		Produces(strings.Split(restMethodConfig.Produces, ",")...).
 		Consumes(strings.Split(restMethodConfig.Consumes, ",")...).
 		Route(ws.Method(restMethodConfig.MethodType).To(rf))
-	grs.container.Add(ws)
+	grs.container.Add(ws)*/
 
+	ws := &restful.WebService{}
+	ws.Path(restServiceConfig.Path)
```

Comment:
It will make user can't define same service path.

Reply:
Config different root path for different service

Reply:
At first, we can't ask user must set server path config. And we should let user more flexible to config path, like spring mvc.

Reply:
You don't have to set the root path. The following configurations are completely equivalent.

```
services:
  "CourseProvider":
    registry: "etcd"
    protocol : "rest"
    interface : "com.llll.Course"
    loadbalance: "random"
    warmup: "100"
    cluster: "failover"
    rest_path: "/courses"
    rest_produces: "application/json;charset=utf-8"
    rest_consumes: "application/json;charset=utf-8,*/*"
    methods:
    - name: "ViewList"
      rest_path: "/"
      rest_method: "GET"
    - name: "ViewOne"
      rest_path: "/{id}"
      rest_path_params: "0:id"
      rest_method: "GET"
    - name: "CreateDoc"
      rest_path: "/"
      rest_body: 0
      rest_method: "POST"
    - name: "EditDoc"
      rest_path: "/{id}"
      rest_path_params: "0:id"
      rest_body: 1
      rest_method: "POST"
```

```
services:
  "CourseProvider":
    registry: "etcd"
    protocol : "rest"
    interface : "com.llll.Course"
    loadbalance: "random"
    warmup: "100"
    cluster: "failover"
    rest_produces: "application/json;charset=utf-8"
    rest_consumes: "application/json;charset=utf-8,*/*"
    methods:
    - name: "ViewList"
      rest_path: "/courses"
      rest_method: "GET"
    - name: "ViewOne"
      rest_path: "/courses/{id}"
      rest_path_params: "0:id"
      rest_method: "GET"
    - name: "CreateDoc"
      rest_path: "/courses"
      rest_body: 0
      rest_method: "POST"
    - name: "EditDoc"
      rest_path: "/courses/{id}"
      rest_path_params: "0:id"
      rest_body: 1
      rest_method: "POST"
```

Reply:
Restful protocol specifies architecture constraints and principles. Usually, a service corresponds to a resource，the rootpath is the resource name.